docs/literate/make.jl

```diff
@@ -0,0 +1,122 @@
+
+    # Generate markdowns
+    function preprocess_docs(content)
+        return string("# # [$title](@id $(splitext(file)[1]))\n $binder_badge\n $nbviewer_badge\n $download_badge\n\n", preprocess_links(content))
+    end
+    Literate.markdown(joinpath(repo_src, folder, file), joinpath(pages_dir, folder); preprocess=preprocess_docs,)
+
+    @testset "TrixiTutorials $title" begin include(joinpath(repo_src, folder, file)) end
+
+end
+
+# Creating tutorials for the following files:
+# Normal structure: "title" => "filename.jl"
+# If there are several files for one topic and folder, the structure is:
+#   "title" => ["subtitle 1" => ("folder 1", "filename 1.jl"),
+#               "subtitle 2" => ("folder 2", "filename 2.jl")]
+files = [
+    "Adding a new equation" => ["Scalar conservation law" => ("adding_new_equations_literate", "cubic_conservation_law_literate.jl"),
+                                "Nonconservative equation" => ("adding_new_equations_literate", "nonconservative_advection_literate.jl")],
+    "Differentiable programming" => "differentiable_programming_literate.jl",
+    ]
+
+repo_src        = joinpath(@__DIR__, "src", "files")
+
+pages_dir       = joinpath(@__DIR__, "..", "src", "tutorials")
+notebooks_dir   = joinpath(@__DIR__, "src", "notebooks")
+
+Sys.rm(pages_dir;       recursive=true, force=true)
+Sys.rm(notebooks_dir;   recursive=true, force=true)
+
+# Preprocess files to add reference web links automatically.
+# trixi_version = Pkg.dependencies()[Pkg.project().dependencies["Trixi"]].version
+trixi_link = "https://trixi-framework.github.io/Trixi.jl/v0.3.61/"
+
+# Function that replaces `@trixi-docs` and `@trixi-ref` with the links to the Trixi documentation
+function preprocess_links(content)
+    # Replacing `@trixi-docs:` in `content` with the defined `trixi_link`
+    content = replace(content, "@trixi-docs:" => trixi_link)
+    # Searching for `[`Example`](@trixi-ref)` in content and replace it with `[`Example`](trixi_link/reference_trixi/#Trixi.Example)`.
+    content = replace(content, r"\[`(?<ref>\w+)`\]\(@trixi-ref\)" => SubstitutionString("[`\\g<ref>`]($(trixi_link)reference-trixi/#Trixi.\\g<ref>)"))
+end
+
+binder_logo   = "https://mybinder.org/badge_logo.svg"
+nbviewer_logo = "https://raw.githubusercontent.com/jupyter/design/master/logos/Badges/nbviewer_badge.svg"
+download_logo = "https://camo.githubusercontent.com/aea75103f6d9f690a19cb0e17c06f984ab0f472d9e6fe4eadaa0cc438ba88ada/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f646f776e6c6f61642d6e6f7465626f6f6b2d627269676874677265656e"
+
+binder_url   = "https://mybinder.org/v2/gh/trixi-framework/TrixiTutorials/gh-pages?filepath=dev/notebooks"
+nbviewer_url = "https://nbviewer.jupyter.org/github/trixi-framework/TrixiTutorials/blob/gh-pages/dev/notebooks"
+download_url = "https://raw.githubusercontent.com/trixi-framework/TrixiTutorials/gh-pages/dev/notebooks"
```

Comment:
You use these global constants only in `create_files`, don't you? If so, it is much cleaner to define them there, avoiding global states etc.